web/client/codechecker_client/cmd_line_client.py

```diff
@@ -320,6 +332,31 @@ def add_filter_conditions(client, report_filter, args):
     if 'fixed_at' in args:
         report_filter.fixDate = int(str_to_timestamp(args.fixed_at))
 
+    detected_at = None
+    fixed_at = None
+
+    if 'detected_before' in args or 'detected_after' in args:
+        detected_at = ttypes.DateInterval()
+
+        if 'detected_before' in args:
+            detected_at.before = int(str_to_timestamp(args.detected_before))
```

Comment:
wouldn't it better to keep the datetime type until it is sent overt thrift? every date related computation, comparison  or modification can be done much easier on the datetime type
with the simple [timestamp](https://docs.python.org/3/library/datetime.html?datetime.datetime.timestamp#datetime.datetime.timestamp) method it can be converted to unix epoch if needed. with the [fromtimestamp](https://docs.python.org/3/library/datetime.html?datetime.datetime.timestamp#datetime.datetime.fromtimestamp) it can be converted back to datetime

Reply:
I you check the `add_filter_conditions` function and check the usages you will see that we give a brand new report filter object to it and it will process the arguments and fill the fields of this report filter. So in this place we actually creating a ReportFilter object which will be sent over the thrift.